env0/data_sshkey.go

```diff
@@ -60,22 +60,25 @@ func getSshKeyByName(name interface{}, meta interface{}) (client.SshKey, diag.Di
 	apiClient := meta.(*client.ApiClient)
 
 	sshKeys, err := apiClient.SshKeys()
-	var sshKey client.SshKey
 	if err != nil {
 		return client.SshKey{}, diag.Errorf("Could not query ssh keys: %v", err)
 	}
-	if len(sshKeys) > 1 {
-		return client.SshKey{}, diag.Errorf("Found multiple SSH Keys for name: %s. Use ID instead or make sure SSH Keys names are unique %v", name, sshKeys)
-	}
+
+	var sshKeysByName []client.SshKey
```

Comment:
@yaronya you are the Go master - is there a normal way we could DRY this code ? 
Something like `getOnlyOneByName` ? 
We do the same thing for projects, ssh keys, templates, maybe more things ...

Reply:
@avnerenv0 Wouldn't say master 😿 
I gave that an effort but found myself struggling with the idea of polymorphism in Go which exists only for `interface`s and not for `struct`s (in this case we need a "parent struct" with a `Name` member).

Since I really wanna finish the UTs today - I let that one go. Do you think it's worth an issue?

Reply:
<div><img src="https://media1.giphy.com/media/3o7TKNRz1oDplVigj6/200.gif?cid=5a38a5a2re72udklm27484gwzvbemt8ekr774z2x1vi0uibs&amp;rid=200.gif&amp;ct=g" style="border:0;height:170px;width:300px"/><br/>via <a href="https://giphy.com/bfv/">Braxton Family Values </a> on <a href="https://giphy.com/gifs/bfv-sisters-braxton-family-values-we-tv-3o7TKNRz1oDplVigj6">GIPHY</a></div>